apps/api/v1/lib/utils/isAdmin.ts

```diff
@@ -1,10 +1,39 @@
 import type { NextApiRequest } from "next";
 
 import prisma from "@calcom/prisma";
-import { UserPermissionRole } from "@calcom/prisma/enums";
+import { UserPermissionRole, MembershipRole } from "@calcom/prisma/enums";
+
+export const ScopeOfAdmin = {
+  SystemWide: "SystemWide",
+  OrgOwnerOrAdmin: "OrgOwnerOrAdmin",
+} as const;
 
 export const isAdminGuard = async (req: NextApiRequest) => {
   const { userId } = req;
-  const user = await prisma.user.findUnique({ where: { id: userId } });
-  return user?.role === UserPermissionRole.ADMIN;
+  const user = await prisma.user.findUnique({ where: { id: userId }, select: { role: true } });
+  if (!user) return { isAdmin: false, scope: null };
```

Comment:
should we throw error here? user not found

Reply:
It's here just to satisfy Typescript really, it will pretty much never happen as `userId` is fetched from the API key in the `verifyApiKey` and it returns not found error if APIKey is invalid. 
![image](https://github.com/calcom/cal.com/assets/52925846/8e8beca7-5ec4-4eae-8600-06f82b52613e)

isAdmin is called after that check, so it would pretty much never satisfy !user here, ever.